pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -1352,6 +1352,13 @@ async function resolveDependency (
         ),
       }
     }
+    if (pkg.peerDependenciesMeta != null) {
+      for (const [peerName, peerOpts] of Object.entries(pkg.peerDependenciesMeta)) {
+        if (peerOpts.optional && (pkg.dependencies?.[peerName] != null || pkg.optionalDependencies?.[peerName] != null)) {
+          peerOpts.optional = false
+        }
+      }
+    }
```

Comment:
This still seems to have problems. For example, in the #8912 issue, the file line 1344 has deleted `lightningcss` in the dependencies. Even if the value is changed here, `lightningcss` will still not be downloaded.